### PR TITLE
Add Intangible Fixed Assets Amortisation Controller

### PR DESCRIPTION
**Pull request by BSaunders95**
Add intangible fixed assets amortisation controller and page and add details to the 'check your answers' page

Resolves SFA-634

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyController.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            LOGGER.errorRequest(request, e.getMessage(), e);
+            return ERROR_VIEW;
+        }
+
+        return Navigator.getNextControllerRedirect(this.getClass(), companyNumber, transactionId, companyAccountsId);
+    }
+
+    @Override
+    protected String getTemplateName() {
+        return "smallfull/intangibleAmortisationPolicy";
```
Comment (Mathsh) [inline src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyController.java]: possibly make this a static final variable?

File: src/main/resources/messages.properties [hunk 2]
```diff
@@ -99,4 +99,8 @@ validation.length.maxExceeded.accounting_policies.basis_of_measurement_and_prepa
 
 validation.characters.invalid.accounting_policies.turnover_policy = Enter valid characters
 validation.length.minInvalid.accounting_policies.turnover_policy = Give details of your 'turnover policy'
-validation.length.invalidInputLength.accounting_policies.turnover_policy = You must not exceed {1} characters
\ No newline at end of file
+validation.length.invalidInputLength.accounting_policies.turnover_policy = You must not exceed {1} characters
+
+validation.characters.invalid.accounting_policies.intangible_fixed_assets_amortisation_policy = Enter valid characters
+validation.length.minInvalid.accounting_policies.intangible_fixed_assets_amortisation_policy = Give details of your ''intangible fixed assets amortisation'' policy
```
Comment (Mathsh) [inline src/main/resources/messages.properties]: is this supposed to be ' 'intangible fixed assets amortisation' '? with double ' '

File: src/main/resources/templates/smallfull/intangibleAmortisationPolicy.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+              legendText = 'Do your prepared accounts include an \'intangible fixed assets amortisation\' policy?',
+              boxLabelTextJavaScriptOn = 'Give details of your \'intangible fixed assets amortisation\' policy',
+              boxLabelTextJavaScriptOff = 'If yes, enter your \'intangible fixed assets amortisation\' policy',
+              radioButtonsFieldName = 'includeIntangibleAmortisationPolicy',
+              textBoxFieldName = 'intangibleAmortisationPolicyDetails',
+              dataTargetTextFieldName = 'intangibleAmortisationPolicy-text-box',
+              textAreaClassName = 'govuk-textarea'
+              )">
+          </div>
+
```
Comment (Mathsh) [inline src/main/resources/templates/smallfull/intangibleAmortisationPolicy.html]: spacing

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+                .thenThrow(ServiceException.class);
+
+        this.mockMvc.perform(postRequestWithValidData())
+                .andExpect(status().isOk())
+                .andExpect(view().name(ERROR_VIEW));
+    }
+
+    private MockHttpServletRequestBuilder postRequestWithValidData() {
+
+        String beanElement = "includeIntangibleAmortisationPolicy";
```
Comment (Mathsh) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java]: "includeIntangibleAmortisationPolicy" should become a static final variable

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        // Mock boolean field input
+        String validData = "1";
+
+        return post(INTANGIBLE_AMORTISATION_POLICY_PATH).param(beanElement, validData);
+    }
+
+    private MockHttpServletRequestBuilder postRequestWithInvalidData() {
+
+        String beanElement = "includeIntangibleAmortisationPolicy";
```
Comment (Mathsh) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java]: same as above

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+        // Mock lack of boolean field input
+        String invalidData = null;
+
+        return post(INTANGIBLE_AMORTISATION_POLICY_PATH).param(beanElement, invalidData);
+    }
+
+
```
Comment (Mathsh) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java]: spacing

**Comment by Mathsh** [comment]
some minor comments, but all else looks good

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java (hunk 6 above)
Comment (BSaunders95) [reply src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java]: Addressed

File: src/main/resources/templates/smallfull/intangibleAmortisationPolicy.html (hunk 3 above)
Comment (BSaunders95) [reply src/main/resources/templates/smallfull/intangibleAmortisationPolicy.html]: Addressed

File: src/main/resources/messages.properties (hunk 2 above)
Comment (BSaunders95) [reply src/main/resources/messages.properties]: Yeah, this is a Thymeleaf thing. The first ' escapes the second to the text reads onscreen as 'intangible fixed assets amortisation'

File: src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyController.java (hunk 1 above)
Comment (BSaunders95) [reply src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyController.java]: I don't think it wins us much, it's another variable assignment for no gain but I'm happy to be guided here if it's a preference?

**Pull request merged**